File: api/api.web.model/GameRequestWebModel.cs
using System;
using System.ComponentModel;

namespace Apex.Api.Web.Model
{
    public class CreatePlayerRequest
    {
        public PlayerKind Kind { get; set; }
        public int? UserId { get; set; }
        public string Name { get; set; }
    }

    public class SelectionRequest
    {
        public int CellId { get; set; }
    }

    public class EventsQuery
    {
        public int? MaxResults { get; set; }
        public ListSortDirection Direction { get; set; }
        public DateTime? ThresholdTime { get; set; }
        public int? ThresholdEventId { get; set; }
    }
}

File: api/api.web/Model/GameRequestWebModel.fs
namespace Apex.Api.Web.Model

open System
open System.ComponentModel
open Apex.Api.Enums
open System.ComponentModel.DataAnnotations

[<CLIMutable>]
type CreatePlayerRequestDto = {
    kind : PlayerKind
    userId : Nullable<int>

    // Nullable
    [<StringLength(20, MinimumLength = 1)>]
    [<RegularExpression("[a-zA-Z0-9\-_]+")>]
    name : string
}

[<CLIMutable>]
type SelectionRequestDto = {
    cellId : int
}

[<CLIMutable>]
type EventsQueryDto = {
    maxResults : Nullable<int>
    direction : ListSortDirection
    thresholdTime : Nullable<DateTime>
    thresholdEventId : Nullable<int>
}